stratum/hal/bin/np4intel/main.cc

```diff
@@ -0,0 +1,190 @@
+              "Path to initial pipeline for Netcope (required for "
+              "starting Netcope)");
+DEFINE_uint32(cpu_port, 128,
+              "Netcope port number for CPU port (used for packet I/O)");
+DEFINE_bool(np4_sim, false, "Run with the NP4 simulator");
+DEFINE_string(dpdk_config, "",
+              "DPDK EAL init config file");
+
+using ::pi::fe::proto::DeviceMgr;
+
+namespace stratum {
+namespace hal {
+namespace np4intel {
+
+namespace {
+
+void registerDeviceMgrLogger() {
+  using ::pi::fe::proto::LogWriterIface;
+  using ::pi::fe::proto::LoggerConfig;
+  class P4RuntimeLogger : public LogWriterIface {
+    void write(Severity severity, const char *msg) override {
+      ::google::LogSeverity new_severity = INFO;
+      switch (severity) {
+        case Severity::TRACE:
+          if (!VLOG_IS_ON(3)) return;
+          new_severity = INFO;
+          break;
+        case Severity::DEBUG:
+          if (!VLOG_IS_ON(1)) return;
+          new_severity = INFO;
+          break;
+        case Severity::INFO:
+          new_severity = INFO;
+          break;
+        case Severity::WARN:
+          new_severity = WARNING;
+          break;
+        case Severity::ERROR:
+          new_severity = ERROR;
+          break;
+        case Severity::CRITICAL:
+          new_severity = FATAL;
+          break;
+      }
+
+      // we use google::LogMessage directly (instead of the LOG macro) so we can
+      // control the file name and line number displayed in the logs. Probably
+      // not ideal but stratum/glue/status/status.h already does the same thing.
+      constexpr char kDummyFile[] = "PI-device_mgr.cpp";
+      constexpr int kDummyLine = 0;
+      google::LogMessage log_message(kDummyFile, kDummyLine, new_severity);
+      log_message.stream() << msg;
+    }
+  };
+  LoggerConfig::set_writer(std::make_shared<P4RuntimeLogger>());
+}
+
+// Initialise the DPDK EAL
+::util::Status  DPDKEalInit() {
+    std::string pgm_name = "stratum_np4intel";
+
+    // Arguments vector
+    std::vector<char*> argv;
+    argv.push_back(strdup(pgm_name.c_str()));
+
+    // Were we passed a dpdk config file
+    ::stratum::hal::np4intel::DPDKConfig dpdk_config;
+    if (!FLAGS_dpdk_config.empty()) {
+        RETURN_IF_ERROR(ReadProtoFromTextFile(FLAGS_dpdk_config,
+                                              &dpdk_config));
+
+        // create argv
+        for (const auto& arg : dpdk_config.eal_args())
+            argv.push_back(strdup(arg.c_str()));
+    }
+    argv.push_back(nullptr);
+
+    if (dpdk_config.disabled()) {
+        LOG(INFO) << "DPDK is disabled";
+    } else {
+        // Call the DPDK EAL init
+        auto rc =  ::pi::np4::DeviceMgr::DPDKInit(argv.size()-1, &argv[0]);
+
+        // Now log the return code message
+        if (rc != 0) {
+            RETURN_ERROR(ERR_INTERNAL) << "DPDK EAL Init failed";
+        }
+        LOG(INFO) << "DPDK EAL Init successful";
+    }
+
+    return ::util::OkStatus();
+}
```

Comment:
Here is an example of 4 spaces.

Reply:
fixed, I've run clang-format over all the new code.